lib/UnexpectedMitmMocker.js

```diff
@@ -0,0 +1,538 @@
+                var mockResponseError = result.error;
+
+                if (!(mockResponse || mockResponseError)) {
+                    return;
+                }
+
+                return Promise.resolve().then(function () {
+                    var assertionMockRequest = new messy.HttpRequest(requestProperties);
+                    trimHeadersLower(assertionMockRequest);
+
+                    expect.errorMode = 'default';
+                    return expect(assertionMockRequest, 'to satisfy', expectedRequestProperties);
+                }).then(function () {
+                    deliverMockResponse(mockResponse, mockResponseError);
+                }).catch(function (e) {
+                    assertMockResponse(mockResponse, mockResponseError);
+                    throw new errors.EarlyExitError('Seen request did not match the expected request.');
+                });
+            }).catch(function (e) {
+                /*
+                 * Given an error occurs, the deferred assertion
+                 * will still be pending and must be completed. We
+                 * do this by signalling the error on the socket.
+                 */
+                try {
+                    clientSocket.emit('error', e);
+                } catch (e) {
+                    /*
+                     * If an something was thrown trying to signal
+                     * an error we have little choice but to simply
+                     * reject the assertion. This is only safe with
+                     * Unexpected 10.15.x and above.
+                     */
+                }
+
+                timeline.push(e);
+                reject(e);
+            });
+
+            function assertMockResponse(mockResponse, mockResponseError) {
+                var mockRequest = new messy.HttpRequest(requestProperties);
+
+                var httpExchange = new messy.HttpExchange({request: mockRequest});
+                if (mockResponse) {
+                    httpExchange.response = mockResponse;
+                } else if (mockResponseError) {
+                    httpExchange.response = mockResponseError;
+                }
+
+                var eventSpec = null;
+                // only attempt request validation with a mock
+                if (hasRequestDescription) {
+                    eventSpec = {request: expectedRequestProperties};
+                }
+
+                /*
+                 * We explicitly do not complete the promise
+                 * when the last request for which we have a
+                 * mock is seen.
+                 *
+                 * Instead simply record any exchanges that
+                 * passed through and defer the resolution or
+                 * rejection to the handler functions attached
+                 * to the delegated assertion execution below.
+                 */
+
+                timeline.push({
+                    exchange: httpExchange,
+                    spec: eventSpec
```

Comment:
Could we find a way to call the property and the variables the same? Would be nice for consistency, and we'll be ready for using the shorthand syntax later :)